typescript-requiring-type-checking.js

```diff
@@ -15,23 +15,23 @@ module.exports = {
         /*
             TypeScript rules outside of the recommended configuration
         */
-        // '@typescript-eslint/naming-convention': 'error',
-        // '@typescript-eslint/no-base-to-string': 'error',
-        // '@typescript-eslint/no-confusing-void-expression': 'error',
-        // '@typescript-eslint/no-unnecessary-boolean-literal-compare': 'error',
-        // '@typescript-eslint/no-unnecessary-condition': 'error',
-        // '@typescript-eslint/no-unnecessary-qualifier': 'error',
-        // '@typescript-eslint/no-unnecessary-type-arguments': 'error',
-        // '@typescript-eslint/non-nullable-type-assertion-style': 'error',
-        // '@typescript-eslint/prefer-includes': 'error',
-        // '@typescript-eslint/prefer-nullish-coalescing': 'error',
-        // '@typescript-eslint/prefer-readonly': 'error',
-        // '@typescript-eslint/prefer-readonly-parameter-types': 'error',
-        // '@typescript-eslint/prefer-reduce-type-parameter': 'error',
-        // '@typescript-eslint/prefer-string-starts-ends-with': 'error',
-        // '@typescript-eslint/promise-function-async': 'error',
-        // '@typescript-eslint/require-array-sort-compare': 'error',
-        // '@typescript-eslint/strict-boolean-expressions': 'error',
-        // '@typescript-eslint/switch-exhaustiveness-check': 'error'
+        '@typescript-eslint/naming-convention': 'error',
+        '@typescript-eslint/no-base-to-string': 'error',
+        '@typescript-eslint/no-confusing-void-expression': 'error',
```

Comment:
This rule requires braces around arrow functions that return void, which I think is unnecessary in many cases.
Example from an observable chain:
 ```
this.propertyChange$.pipe(
    filter((): boolean => this.canLoadIframe()),
    filter((changes: SimpleChanges): boolean => this.changeRequiresReload(changes)),
    switchMap((): Observable<Document> => this.getDocument()),
    tap((document: Document): Document => this.document = document),
    tap(() => this.updateIframe())
).subscribe()
```

This rule will error on `tap(() => this.updateIframe())` and wants it to be something like:
```
this.propertyChange$.pipe(
    filter((): boolean => this.canLoadIframe()),
    filter((changes: SimpleChanges): boolean => this.changeRequiresReload(changes)),
    switchMap((): Observable<Document> => this.getDocument()),
    tap((document: Document): Document => this.document = document),
    tap(() => {
        this.updateIframe()
    })
).subscribe()
```

**Proposal**
```
"@typescript-eslint/no-confusing-void-expression": [
    "error",
    { "ignoreArrowShorthand": true }
]
```

Reply:
This has the votes needed